src/libstore/gc.cc

```diff
@@ -786,8 +786,24 @@ void LocalStore::collectGarbage(const GCOptions & options, GCResults & results)
         for (auto & i : options.pathsToDelete) {
             assertStorePath(i);
             tryToDelete(state, i);
-            if (state.dead.find(i) == state.dead.end())
-                throw Error(format("cannot delete path '%1%' since it is still alive") % i);
+            if (state.dead.find(i) == state.dead.end()) {
+                /* We cannot delete the current path: it is referenced by
+                   at least an intermediate path linked to a GC root.
```

Comment:
This assumption is not correct. The path might be a GC root itself, or be a temporary / in-memory root. In either case it does not necessarily have a referrer.